lib/embed-docs/embedDocs.ts

```diff
@@ -225,10 +225,11 @@ export function getDocsLoader(
 export function splitTextByHeaders(
   markdownText: string,
 ): Record<string, string> {
-  const regex = /(#{1,3}) (.+?)\n([\s\S]*?)(?=\n#{1,3} |$)/g;
+  const regex = /(#{1,3}) (.*?)\n([\s\S]*?)(?=\n#{1,3} |$)/g;
   let match;
   let results: Record<string, string> = {};
   let isFirstChunk = true;
+  let prevHeading = "";
```

Comment:
lil comment to say this deals with empty headings would be 💯

Reply:
```suggestion
  // This is to deal with empty headings (so we can use the value of the previous non-empty heading)
  let prevHeading = "";
```